src/Cli/dotnet/commands/dotnet-test/Program.cs

```diff
@@ -70,27 +79,45 @@ public static TestCommand FromParseResult(ParseResult result, string[] settings,
             if (!hasRootVariable)
             {
                 testCommand.EnvironmentVariable(rootVariableName, rootValue);
+                VSTestTrace.WriteTrace($"Root variable set {rootVariableName}:{rootValue}");
             }
 
+            VSTestTrace.SafeWriteTrace(() => $"Starting test using MSBuild with arguments '{testCommand.GetArgumentsToMSBuild()}' custom MSBuild path '{msbuildPath}' norestore '{noRestore}'");
```

Comment:
Why do you use `SafeWriteTrace` but regular `WriteTrace` in the previous if?

Reply:
`SafeWriteTrace` is used only where we don't simply generate a string but also where to do that we're running complex code(where parameters are not primitive). This code will run always so I don't want to fail normal run(where trace are disabled) because for instance in this case `testCommand` is in "weird" strate and `GetArgumentsToMSBuild` will fail.